src/routes/backup/BulkCreateStandbyVolumeModal.js

```diff
@@ -114,30 +90,52 @@ const modal = ({
   const allFieldsError = { ...getFieldsError() }
   const hasFieldsError = Object.values(allFieldsError).some(fieldError => fieldError !== undefined) || false
 
+  const updateDrVolumeConfigs = (key, newValue) => {
+    setDrVolumeConfigs(prev => {
+      const newConfigs = [...prev]
+      const data = {
+        ...getFieldsValue(),
+        [key]: newValue,
+      }
+      newConfigs.splice(currentTab, 1, data)
+      return newConfigs
+    })
+  }
+
+  const handleNameChange = (e) => updateDrVolumeConfigs('name', e.target.value)
+  const handleReplicasNumberChange = (newNumber) => updateDrVolumeConfigs('numberOfReplicas', newNumber)
+  const handleEncryptedCheck = (e) => updateDrVolumeConfigs('encrypted', e.target.checked)
+  const handleDataEngineChange = (value) => updateDrVolumeConfigs('dataEngine', value)
+  const handleAccessModeChange = (value) => updateDrVolumeConfigs('accessMode', value)
+  const handleNodeTagRemove = (value) => {
+    const oldNodeTags = drVolumeConfigs[currentTab]?.nodeSelector
+    const newNodeSelector = oldNodeTags?.filter(tag => tag !== value) || []
+    updateDrVolumeConfigs('nodeSelector', newNodeSelector)
+  }
+  const handleNodeTagAdd = (value) => {
+    const oldNodeTags = drVolumeConfigs[currentTab]?.nodeSelector
+    updateDrVolumeConfigs('nodeSelector', [...oldNodeTags, value])
+  }
+  const handleDiskTagRemove = (value) => {
+    const oldDiskTags = drVolumeConfigs[currentTab]?.diskSelector
+    const newDiskSelector = oldDiskTags?.filter(tag => tag !== value) || []
+    updateDrVolumeConfigs('diskSelector', newDiskSelector)
+  }
+  const handleDiskTagAdd = (value) => {
+    const oldDiskTags = drVolumeConfigs[currentTab]?.diskSelector
+    updateDrVolumeConfigs('diskSelector', [...oldDiskTags, value])
+  }
 
   const handleTabClick = (key) => {
     if (hasFieldsError) {
-      message.error('Please fill in all required fields before switching to another disaster recover volume tab', 5)
+      message.error('Please correct the error fields before switching to another tab', 5)
       return
     }
     validateFields((errors) => {
-      if (errors) {
-        return
-      }
-      const data = {
-        ...getFieldsValue(),
```

Comment:
Previous implementation I choose to set current configuration tab in onChange / onSelect callback by using `getFieldsValue()`.

But that has problem not get the latest updated value in the action callback